lua/obsidian/command.lua

```diff
@@ -168,13 +168,73 @@ command.search = function(client, data)
   end
 end
 
+--- Follow link under cursor.
+---
+---@param client obsidian.Client
+command.follow = function(client, _)
+  local scan = require "plenary.scandir"
+
+  local open, close = util.cursor_on_markdown_link()
+  local current_line = vim.api.nvim_get_current_line()
+
+  if open == nil or close == nil then
+    return nil
+  end
+
+  local note_name = current_line:sub(open + 2, close - 1)
+  local path = client.dir
+  
+  if not note_name:match("%.md") then
+    note_name = note_name .. ".md"
+  end
+  
+  if note_name:match("|[^%]]*") then
+    note_name = note_name:sub(1, note_name:find('|'))
+  end
+
+  local notes = {}
+  local count = 1
+
+  if not note_name:match("/") then
+    scan.scan_dir(vim.fs.normalize(tostring(client.dir)), {
```

Comment:
I think it would be a lot more efficient to scan only directories (use `only_dirs = true`) and then check if the given note name is in each directory.

Alternatively, we don't necessarily need to check every directory. We can just check the same directories that are added to `path`: https://github.com/epwalsh/obsidian.nvim/commit/f08be13e846c35efb0afff37b1ed00033374013c#diff-f3def1404e2ac8d33b653d93273b328b44defe45491bf8589b7598809cb6b854

Reply:
Does the notes_subdir config allow the user to specify multiple subdirectories? 

It makes sense that they might want to dump all new notes into a single folder when creating them, but they might have multiple top level folders to actually store notes in which they may want to link to.

Reply:
> It makes sense that they might want to dump all new notes into a single folder when creating them, but they might have multiple top level folders to actually store notes in which they may want to link to.

True. In that case please ignore the second thing I said, but "I think it would be a lot more efficient to scan only directories (use only_dirs = true) and then check if the given note name is in each directory" is still relevant.

Reply:
Should be done!